django/core/management/commands/makemessages.py

```diff
@@ -382,7 +382,11 @@ def handle(self, *args, **options):
             potfiles = self.build_potfiles()
 
             # Build po files for each selected locale
+            valid_locale = re.compile(r'^[a-z]{1,8}((_{1})([A-Z]{1})'
+                                      r'([a-z]{1,8}|[A-Z]{1,2}))?(_([A-Z]{1,4}|[0-9]{2,4}))?$')
```

Comment:
8 also seems too high here.

Reply:
Nice job catching `ar_001` but I'm concerned about regex bloat to catch a single case.

Reply:
Actually, in `locale_code_re`, the length was 8. So I  thought there could be a locale with 8 lengths. Shall I change it to 3 as I think the maximum length of a language code could be 3?

Reply:
3 sounds good to me. I think you're referring to `language_code_re`, not `locale_code_re`, which doesn't exist. As Claude mentioned in [this comment](https://code.djangoproject.com/ticket/29712#comment:37), that's really why a new regex is needed.